YakimS/assignment4/IslandsCounterTest.java

```diff
@@ -0,0 +1,88 @@
+package com.shaya.CodeUAss4Package;
+
+import org.junit.Assert;
+
+import static org.junit.Assert.*;
+
+public class IslandsCounterTest {
+
+    @org.junit.Test
+    public void countIslandInExampleGrid()
+    {
+        IslandsCounter islandsCounter = new IslandsCounter();
+
+        boolean[][] boolGrid = new boolean[][]
+                {
+                        { false , true, false, true},
+                        { true , true, false, false},
+                        { false , false, true, false},
+                        { false , false, true, false},
+                };
+
+        Assert.assertEquals(islandsCounter.countIsland(boolGrid), 3);

```

Comment:
`assertEquals` takes expected value as first parameter, not the second one. This might seem to be not important, since assertion will fail in case of error no matter what order parameters are in, but will make error message confusing.

Also: no need for `Assert.` part - all methods are already imported.

Reply:
Wow, nice catch!